src/app/portlets/content-types/fields/content-type-fields-variables/content-type-fields-variables.component.ts

```diff
@@ -66,23 +66,51 @@ export class ContentTypeFieldsVariablesComponent implements OnInit, OnChanges, O
      * @param {FieldVariable} variable
      * @memberof ContentTypeFieldsVariablesComponent
      */
-    deleteVariable(fieldIndex: number): void {
-        const params: FieldVariableParams = {
-            contentTypeId: this.field.contentTypeId,
-            fieldId: this.field.fieldId,
-            variable: this.fieldVariables[fieldIndex]
-        };
-        this.fieldVariablesService.delete(params).pipe(take(1)).subscribe(
-            () => {
-                this.fieldVariables = this.fieldVariables.filter(
-                    (_item: FieldVariable, index: number) => index !== fieldIndex
-                );
-                this.canSaveFields.splice(fieldIndex, 1);
-            },
-            (err: ResponseView) => {
-                this.dotHttpErrorManagerService.handle(err).pipe(take(1)).subscribe();
-            }
-        );
+    deleteVariable(fieldVariable: FieldVariable, fieldIndex: number): void {
+        if (fieldVariable.key !== '') {
```

Comment:
Can we call this method when the key of a variable is empty? we shouldn't have field variables with empty keys.

Reply:
?